### ISSUE TITLE
Ignore papers with empty abstract

### ISSUE THREAD

**Issue by pkraker** [OpenKnowledgeMaps/Headstart#101]
Suggestion from @brembs: ignore papers with empty abstracts as they just don't have enough information for analysis. I think that this may improve certain maps quite a bit, while only relinquishing articles with little information, such as editorials. 

We would need to do that at query time, however, as we do not have the time to check the result for empty abstracts and re-query. In PubMed, we can probably do this on based on article types; for DOAJ, where we have no article types, we would need to check whether the abstract is empty. This is not straightforward, esp. since wildcards are disabled in the DOAJ API; see http://stackoverflow.com/questions/5384423/lucene-not-null-query for pointers.

**Comment by chreman** [comment]
do the APIs allow for 100+ results? e.g. query for 200 and only keep the top100 with abstracts. but that doubles initial bandwith use and still does not guarantee full set of papers with abstracts...

**Comment by pkraker** [comment]
@chreman: as you pointed out, there are certain downsides to this practice, which is why I see it as a sort of last resort. 

But I think I found the answer: as DOAJ does not include empty fields in the result, we can use [\_exists\_](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#_field_names) as a sort of filter. See e.g. https://doaj.org/api/v1/search/articles/title%3Aeditorial%20AND%20_exists_%3Abibjson.abstract?pageSize=100

**Comment by Bubblbu** [comment]
I will have a look at the suggestion from @pkraker for DOAJ and see how that goes...

@sckott do you think we could implement this for PubMed without using additional queries?

**Comment by sckott** [comment]
i'll have a look

**Comment by sckott** [comment]
Looks like it can be done with the keywor `hasabstract`

e.g,. a web gui search https://www.ncbi.nlm.nih.gov/pubmed?term=neoplasms%20AND%20hasabstract 

and in R

```r
library(rentrez)
library(xml2)
res <- entrez_search(db = "pubmed", term = "ecology AND hasabstract", sort="relevance", retmax = 200)
out <- entrez_fetch(db = "pubmed", id = res$ids, rettype = "xml")
xml <- xml_children(read_xml(out))
xml_find_all(xml, "//Abstract")
```

**Comment by Bubblbu** [comment]
@sckott thx! looks promising

@pkraker has solved the problem for the BASE integration: 7ab3045f038b796ec35a285956c2683cdacebef9
